### ISSUE TITLE
setRoot from sidemenu cause freeze of view on iOS 12.2

### ISSUE THREAD

**Issue by filippodicostanzo** [ionic-team/ionic-v3#1004]
**I'm submitting a ...**  (check one with "x")
[ X] bug report
[ ] feature request

**Current behavior:**
After some tests the problem continues with the freeze of the views in my IONIC application.
With Ionic Blank application with side menu if open the menu and i select list the list view doesn't scroll. The scroll works only after any events occurs.
If i change the method from setRoot to push the scroll works
The problem occurs only on iOS 12.2 on real device.

**Expected behavior:**
When I change the navigation root the application should work properly

**Steps to reproduce:**
ionic start v3 with sidemenu
ionic cordova platform add ios
ionic cordova build ios
run on real device

**Comment by fabmesto** [comment]
when i open and chance option in ion-select the view freeze

**Comment by mchusma** [comment]
We are also seeing this issue, it seems to be a critical one, causing widespread issues.

**Comment by andrefa** [comment]
Noticed this today as well. Scroll is not working per default, only after minimizing/restoring the app and after some back and forth inside the app.

My environment, if it may help
``` 
cli packages: (/data/projects/tidy/new_hk_app/node_modules)

    @ionic/cli-utils  : 1.19.1
    ionic (Ionic CLI) : 3.19.1

global packages:

    cordova (Cordova CLI) : 7.1.0 

local packages:

    @ionic/app-scripts : 3.1.8
    Cordova Platforms  : android 6.3.0 browser 5.0.3 ios 4.5.5
    Ionic Framework    : ionic-angular 3.9.2

System:

    Android SDK Tools : 26.1.1
    ios-deploy        : 1.9.2 
    ios-sim           : 6.1.2 
    Node              : v7.9.0
    npm               : 5.6.0 
    OS                : macOS
    Xcode             : Xcode 10.1 Build version 10B61 

Environment Variables:

    ANDROID_HOME : /Users/andrealmeida/Library/Android/sdk
```

**Comment by filippodicostanzo** [comment]
@liamdebeasi in virtual-scroll.js
i've founded

```
    VirtualScroll.prototype.scrollUpdate = function (ev) {
        var _this = this;
        // ensure no empty are processed
        if (!ev) {
            return;
        }
        // set the scroll top from the scroll event
        this._data.scrollTop = ev.scrollTop;
        // there is a queue system so that we can
        // spread out the work over multiple frames
        var queue = this._queue;
        if (queue === 1 /* NoChanges */) {
            // no dom writes or change detection to take care of
            this._stepNoChanges();
        }
        else if (queue === 2 /* ChangeDetection */) {
            this._dom.write(function () { return _this._stepChangeDetection(); });
        }
        else {
            (void 0) /* assert */;
            // there are DOM writes we need to take care of in this frame
            this._dom.write(function () { return _this._stepDOMWrite(); });
        }
    };
```

**Comment by liamdebeasi** [comment]
Ok this seems similar to the older iOS 12.2 scrolling bug I fixed when it was beta. I'll do some investigation and report back 👍

**Comment by filippodicostanzo** [comment]
In this topic on ionic forum you can find more informations:

https://forum.ionicframework.com/t/ionic-v3-scroll-issue-after-update-ios-12-2/160826/11

Thank You

**Comment by xmarkclx** [comment]
Can confirm.

As a workaround, any event inside the app fixes the scroll issue.

So putting a button for example, and then programmatically clicking that button helps.

Just add a button with a class button_hidden.

Example:

```
<button class="button_hidden" style="margin: 0; padding: 0; border: 0; height: 0;"></button>
```

```
export default function scrollFix() {
    setTimeout(function () {
        const el = document.getElementsByClassName('button_hidden');
        for (var i = 0; i < el.length; i++) {
            const a: HTMLElement = el[i] as HTMLElement;
            a.click();
        }
        console.info('scroll fixed');
    }, 100);

    setTimeout(function () {
        const el = document.getElementsByClassName('button_hidden');
        for (var i = 0; i < el.length; i++) {
            const a: HTMLElement = el[i] as HTMLElement;
            a.click();
        }
        console.info('scroll fixed');
    }, 750);
}
```

You can make a component for it as well, as I did and just add the component on all the pages.

**Comment by liamdebeasi** [comment]
Hi everyone,

I pushed a new nightly build of Ionic 3 moments ago that has a fix for the scrolling issue. (the other nightly fix I mentioned previously missed a few scenarios).

Please give it a try and let me know if this fixes your issue! `npm i ionic-angular@nightly`. You should end up with `ionic-angular@3.9.4-201904091825` installed.

Thanks!

**Comment by filippodicostanzo** [comment]
@liamdebeasi i tried my app now and it works.
There isn't a scroll bug and i don't use any work around.

Just install the nightly version of ionic-angular.

I've tested on real device with ios 12.2.

Thank You

**Comment by aSchimp** [comment]
@liamdebeasi The new nightly build fixed the scroll bug for me also. But, there needs to be a null check for enteringView because it's null when an overlay closes, which is causing an error and some glitches.

```
if (enteringView.getIONContentRef()) {
....
}
```

should be

```
if (enteringView && enteringView.getIONContentRef()) {
....
}
```

**Comment by liamdebeasi** [comment]
Good catch! I will fix and push out a new nightly.

**Comment by longgt** [comment]
@liamdebeasi if this bug only happens in iOS, why not use `isIos` in `src\platform\platform-utils` to add a ios check for this workaround?
Your temporary fix will affect not only iOS but also Android.

**Comment by Srinivas-Tranzeal** [comment]
@liamdebeasi - Are you able to resolve and push new code?

**Comment by liamdebeasi** [comment]
Hi everyone,

I've added the undefined and iOS checks. A new nightly has been published.

Please try it out and let me know if it resolves your issue.

Thanks!

**Comment by aSchimp** [comment]
@liamdebeasi the `3.9.4-201904101244` nightly build is working great. Thanks!

**Comment by jjacobs33589** [comment]
It still isn't working for us

**Comment by liamdebeasi** [comment]
Hi @jjacobs33589,

Can you provide a repository that I can use to reproduce the issue?

**Comment by JoeMeeks** [comment]
Just add this to your scss:
```
ion-app.platform-ios12 ion-content { pointer-events: auto; /*fix scroll lock on ios12.2*/ }
```

**Comment by liamdebeasi** [comment]
Hi @JoeMeeks,

Adding that CSS breaks some functionality where we need to disable pointer events. For example, when opening the side menu, you should not be able to scroll the little bit of content you can see on the side. If you set `pointer-events: auto` the main content is always scrollable, even when the side menu is open.

**Comment by liamdebeasi** [comment]
Hi everyone,

This issue has been resolved and is available in Ionic 3.9.5.

You can upgrade using `npm i ionic-angular@latest`.

Thanks for your patience as we worked to resolve this issue! If you encounter any more iOS 12.2 issues, please [open a new issue](https://github.com/ionic-team/ionic-v3/issues/new).

**Comment by JoeMeeks** [comment]
@liamdebeasi thanks for calling out the expected functionality of the open side menu. My workaround does however solve the bug that occurs when closing the menu by sacrificing disabling scrolling when it is open. A negligible trade-off for my Ionic 3.9.2 app update which is going to be released next week to address this bug currently happening to iOS 12.2 users in production.

**Comment by JoeMeeks** [comment]
@liamdebeasi this is still happening intermittently after the v9.3.5 upgrade:

1. Open the side menu from the home page
2. Close the side menu by tapping the menu icon button
3. Home page no longer scrolls
When navigating to other pages via the side menu the destination pages scroll okay. Afterwards returning to the home page will work and the home page scrolls again. The issue is then reproducible by doing steps 1–3 again. So basically dismissing the menu with no page change on my navigation root page is causing the problem.

The only way for me to completely eliminate the issue is to use my CSS workaround which does solve the bug that occurs when closing the menu by sacrificing disabling of scrolling when it is open. A negligible trade-off for my Ionic 3.9.5 app update which is going to be released next week to address this bug currently happening to iOS 12.2 users in production.

**Comment by liamdebeasi** [comment]
Hi @JoeMeeks,

I am unable to reproduce this using the Ionic Conference App. Can you provide a repository with the code required to reproduce this issue?

Thanks!

**Comment by JoeMeeks** [comment]
@liamdebeasi apologies for the delay but I am working on a repro repo now.

**Comment by jayordway** [comment]
I am experiencing this with ionic 4 and capacitor. Any suggestions, are these fixes just going into ionic 3?

**Comment by liamdebeasi** [comment]
Hi @jayordway,

If you are experiencing this issue with Ionic 4 please create an issue in the v4 repo here: https://github.com/ionic-team/ionic/issues 

Getting this to work properly on both v3 and v4 is a priority for us, so I will take a look once the issue is filed. Also if you have a code reproduction that'd be helpful as well.

Thanks!

**Comment by JoeMeeks** [comment]
@liamdebeasi  I stripped down my project to the bare essentials and made this repo:
https://github.com/JoeMeeks/v3SideMenu-iOS12.2-bug

**Comment by jayordway** [comment]
@liamdebeasi Now that I have add ion-content `{ pointer-events: auto; /*fix scroll lock on ios12.2*/ }`
to my global.scss the problem is no longer an issue in my ionic 4 app.

**Comment by liamdebeasi** [comment]
Hi @jayordway,

Can you provide a repository with this issue occurring in an Ionic 4 app? I am glad to hear the issue is fixed for you with that patch, but if the problem still exists in Ionic 4 I would like to get this resolved for everyone.

**Comment by JoeMeeks** [comment]
@liamdebeasi are you able to reproduce the issue with  my v3SideMenu-iOS12.2-bug repo?

**Comment by liamdebeasi** [comment]
Hi @JoeMeeks,

I am able to reproduce. The solution seems similar to the one I pushed out. Setting a z-index on `ion-nav` and then removing it seems to cause a reflow in WKWebView, bringing back pointer-events.

I will post here when I have a fix in place.

Thanks!

**Comment by lsantaniello** [comment]
Sorry guys, I'm a bit confused. My app use ionic3, could you please share with my the final fix to solve the issue?

Thanks

**Comment by Mateoc** [comment]
Hi
@liamdebeasi I'm getting this on v4, I can provide an example if needed (ios 12.3)

**Comment by liamdebeasi** [comment]
@Mateoc,

Could you file an issue here: https://github.com/ionic-team/ionic/issues

This is the issue tracker we use for v4. Also, please provide an example - that will make it easier for us to test.

Thanks!

**Comment by AaronChengHao** [comment]
@liamdebeasi  Hello, could you please tell me if the sidebar skip home page can't slide? Has the problem been solved?
